src/components/chats/ChatList/hooks/useGetChatElement.ts

```diff
@@ -0,0 +1,130 @@
+import useWrapCallbackInRef from '@/hooks/useWrapCallbackInRef'
```

Comment:
In generaly it is really unclear code, need to load all refs to brain for understanding this hook

Explain your algorithm for loading comment when you wait replyComment, maybe I can propose do it simpler :)

Reply:
The main goal for this is to generate a simple async function, that returns the chat element (using `getElementById`).

But the problem is, if the chat element is not rendered yet, I need to make a `loadMore` calls, until it's rendered.
To do so, I create something like a hack to make the component who uses this hook don't need to care about any of it, it just waits the function to return.

The flow:
1. component calls `await getChatElement(id)`
2. it will generate new promise, which will be awaited by that function, and send the resolver to the ref containing all resolvers.
3. in the meantime, the usual `useEffect` will listen to the changes of the loaded comments, and resolve promises which are waiting for those commentIds.
4. If the promise is resolved, then the function call now waits until the loading for all chats are done, and only after that the element is returned.

Reply:
I see, but add a comment that tries to rewrite it to be more elegant; I dont want to invest time in it right now

Reply:
Done in [85a0e79](https://github.com/dappforce/grillchat/pull/38/commits/85a0e799831aca70814d15ff7ddb5ed33a378cae)